tests/scripts/auto-grow-storage.sh

```diff
@@ -0,0 +1,134 @@
+         #     newsize = sum(node_filesystem_size_bytes)
+         #     echo "Disk has reached it's MAX capacity, add a new disk to it"
+   result=$(oc patch pvc ${pvc} -n ${ns} --type json --patch  "[{ "op": "replace", "path": "/spec/resources/requests/storage", "value": "${newsize}${unitsize}" }]")
+   echo ${result}
+}
+
+function growHorizontally() {
+   count=$1
+   pvc=$2
+   ns=$3 
+   deviceSetName=$(oc get pvc  ${pvc}  -n ${ns}  -o json | jq -r '.metadata.labels."ceph.rook.io/DeviceSet"')
+   cluster=""
+   deviceSets=""
+   currentCount=0
+   clusterCount=0
+   deviceSetCount=0
+   while [ -z "$cluster" ]
+   do
+      cluster=$(oc get CephCluster ${ns} -o json | jq -r ".items["$clusterCount"]")
+      while  [ -z "$deviceSets" ]
+      do
+         deviceSet=$(oc get CephCluster ${ns} -o json | jq -r ".items[$clusterCount].spec.storage.storageClassDeviceSets[$deviceSetCount].name")
+         if [[ $deviceSet == $deviceSetName ]]
+         then
+            currentCount=$(oc get CephCluster ${ns} -o json | jq -r ".items[$clusterCount].spec.storage.storageClassDeviceSets[$deviceSetCount].count")
+            finalCount=$(( $currentCount + $count ))
+            echo "OSDs count will be increased by ${count},Total count for device set ${deviceSetName} is ${finalCount}"
+            result=$(oc patch CephCluster rook-ceph ${ns} --type json --patch "[{ "op": "replace", "path": "/spec/storage/storageClassDeviceSets/${deviceSetCount}/count", "value": "${finalCount}" }]")
+            echo ${result}
+            break
+         fi
+         deviceSetCount=$((deviceSetCount+1))    
+      done
+      currentCount=$((currentCount+1))
+   done
+}
+
+function growOSD(){
+   i=0
+   alertmanagerroute=$(oc get route -n openshift-monitoring | grep alertmanager-main | awk '{ print $2 }')
+   curl -sk -H "Authorization: Bearer $(oc sa get-token prometheus-k8s -n openshift-monitoring)"  https://${alertmanagerroute}/api/v1/alerts | jq -r '.' >./tt.txt
+   export total_alerts=$(cat ./tt.txt | jq '.data | length')
+   echo "Looping at $(date +"%Y-%m-%d %H:%M:%S")"
+
+   while true
+   do
+      export entry=$(cat ./tt.txt | jq ".data[$i]")
+      thename=$(echo $entry | jq -r '.labels.alertname')
+      if [ x"${thename}" = "xPersistentVolumeUsageNearFull" ] || [ x"${thename}" = "xPersistentVolumeUsageCritical" ]
+      then
+         echo $entry
+         ns=$(echo $entry | jq -r '.labels.namespace')
+         pvc=$(echo $entry | jq -r '.labels.persistentvolumeclaim')
+         echo "Processing NearFull or Full alert for PVC ${pvc} in namespace ${ns}"
+         if [[ $1 == "count" ]]
+         then 
+            growHorizontally $2 ${pvc} ${ns}
+         else
+            growVertically $2 ${pvc} ${ns}
+         fi
+
+      (( i = i + 1 ))
+      if (( i == total_alerts ))
+      then
+         sleep 300
+         rm -f ./tt.txt
+         alertmanagerroute=$(oc get route -n openshift-monitoring | grep alertmanager-main | awk '{ print $2 }')
```

Comment:
yes, the Prometheus metrics and alerts are auto-configured with OCP.

Reply:
Since this is an example script, it's a fair question whether it's worth the effort for it to work on both K8s and OCP. First preference is for examples with K8s, but since the example was provided first with OCP, perhaps we leave it this way for this PR and then plan on a separate PR for K8s.